src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/CommonHelper.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class CommonHelper
+    {
+        public static bool IsArrayEqual<T>(T[] array1, T[] array2)
+        {
+            if (array1.Length != array2.Length)
+            {
+
+                return false;
+            }
+
+            for (int i = 0; i < array1.Length; i++)
+                if (!array1[i].Equals(array2[i]))
+                {
+
+                    return false;
+                }
+
+            return true;
+        }
```

Comment:
Could this whole helper be:
```C#
public static bool IsArrayEqual<T>(T[] array1, T[] array2) => array1.AsSpan().SequenceEqual(array2);
```
?

Reply:
It seems SequenceEqual does not work with T.
Error	CS0314	The type 'T' cannot be used as type parameter 'T' in the generic type or method 'MemoryExtensions.SequenceEqual<T>(Span<T>, ReadOnlySpan<T>)'. There is no boxing conversion or type parameter conversion from 'T' to 'System.IEquatable<T>'.

Reply:
Can you constrain T?
```C#
public static bool IsArrayEqual<T>(T[] array1, T[] array2) where T : IEquatable<T> => array1.AsSpan().SequenceEqual(array2);
```
?

Reply:
This works fine, thanks!